buildbot/start_script.sh

```diff
@@ -7,170 +7,107 @@
-        | bash
-    ) && exit 0
+    ${SCRIPT_DIR}/../hwaddress-sanitizer/create_qemu_image.sh && exit 0
   done
   exit 1
 ) || $ON_ERROR
 
-SERVICE_NAME=buildbot-worker@b.service
-[[ -d /var/lib/buildbot/workers/b ]] || ln -s $BOT_DIR /var/lib/buildbot/workers/b
-
-systemctl enable $SERVICE_NAME
-systemctl set-property $SERVICE_NAME TasksMax=100000
-
-systemctl stop $SERVICE_NAME || true
-while pkill buildbot-worker; do sleep 5; done;
-
-rm -f /b/buildbot.tac
-buildbot-worker create-worker -f --allow-shutdown=signal $BOT_DIR lab.llvm.org:$MASTER_PORT \
-  "sanitizer-$(hostname | cut -d '-' -f2)" \
-  "$(gsutil cat gs://sanitizer-buildbot/buildbot_password)"
-
-
-echo "Vitaly Buka <vitalybuka@google.com>" > $BOT_DIR/info/admin
-
-{
-  echo "How to reproduce locally: https://github.com/google/sanitizers/wiki/SanitizerBotReproduceBuild"
-  echo
-  uname -a | head -n1
-  date
-  cmake --version | head -n1
-  g++ --version | head -n1
-  ld --version | head -n1
-  lscpu
-} > $BOT_DIR/info/host
-
-mkdir -p $BOT_DIR/.ccache
-mkdir -p /var/lib/buildbot/.ccache
-cat <<EOF >/var/lib/buildbot/.ccache/ccache.conf
-max_size = 20.0G
-compression = true
-cache_dir = $BOT_DIR/.ccache
-EOF
-
-chown -R buildbot:buildbot $BOT_DIR
-systemctl daemon-reload
-systemctl start $SERVICE_NAME
-systemctl status $SERVICE_NAME
-
-sleep 30
-cat $BOT_DIR/twistd.log
-grep "worker is ready" $BOT_DIR/twistd.log || $ON_ERROR
-
-# GCE can restart instance after 24h in the middle of the build.
-# Gracefully restart before that happen.
-sleep 72000
-while pkill -SIGHUP buildbot-worker; do sleep 5; done;
-$ON_ERROR
+function create_worker() {
+  local WORKER_NAME="$1"
+  local SERVICE_NAME=buildbot-worker@b.service
+  [[ -d /var/lib/buildbot/workers/b ]] || ln -s $BOT_DIR /var/lib/buildbot/workers/b
+
+  systemctl enable $SERVICE_NAME
+  systemctl set-property $SERVICE_NAME TasksMax=100000
+
+  systemctl stop $SERVICE_NAME || true
+  while pkill buildbot-worker; do sleep 5; done;
+
+  rm -f ${BOT_DIR}/buildbot.tac ${BOT_DIR}/twistd.log
+  buildbot-worker create-worker -f --allow-shutdown=signal $BOT_DIR lab.llvm.org:$SERVER_PORT \
+    "$WORKER_NAME" \
+    "$(gsutil cat gs://sanitizer-buildbot/buildbot_password)"
+
+  echo "Vitaly Buka <vitalybuka@google.com>" > $BOT_DIR/info/admin
+
+  {
+    echo "How to reproduce locally: https://github.com/google/sanitizers/wiki/SanitizerBotReproduceBuild"
+    echo
+    uname -a | head -n1
+    date
+    cmake --version | head -n1
+    g++ --version | head -n1
+    ld --version | head -n1
+    lscpu
+  } > $BOT_DIR/info/host
+
+  chown -R buildbot:buildbot $BOT_DIR
+  systemctl daemon-reload
+  systemctl start $SERVICE_NAME
+  systemctl status $SERVICE_NAME
+  sleep 30
+  cat ${BOT_DIR}/twistd.log
+  grep "worker is ready" $BOT_DIR/twistd.log
+}
+
+function is_worker_connected() {
+  local WORKER_NAME="$1"
+  (
+    set -o pipefail
+    curl http://lab.llvm.org:${API_PORT}/api/v2/workers/${WORKER_NAME} \
```

Comment:
It's probably browser redirects you to https
Try in Incognito browser http://lab.llvm.org:8014/api/v2/workers/sanitizer-buildbot1
I don't mind to use https, but I am not sure what is the port for staging, we need easy ability to switch prod/staging
Probably I can use API_URL instead of API_PORT

Reply:
done